### ISSUE TITLE
Bug: <select> element with static value but async options may have unexpected behavior.

### ISSUE THREAD

**Issue by j-sen** [facebook/react#17826]
React version:  16.12.0

I'm not sure if this is a bug or an acceptable behavior.

## Steps To Reproduce

1. Render a &lt;select&gt; element.
2. Pass a value or defaultValue to it.
3. Render it's &lt;option&gt; elements  asynchronously.

Link to code example: https://codesandbox.io/s/select-element-with-static-value-but-async-options-l3hns

## The current behavior

&lt;select&gt; element with static value but async options ignores it's value or defaultValue props until the &lt;select&gt; element get re-rendered.

## The expected behavior

&lt;select&gt; element should not ignores it's value or defaultValue props silently.

**Comment by kunukn** [comment]
I think the shown code and behaviour is as expected.

The documentation about `defaultValue` says

>... you often want React to specify the initial value, but leave subsequent updates uncontrolled. 
>To handle this case, you can specify a defaultValue attribute instead of value.

https://reactjs.org/docs/uncontrolled-components.html#default-values

```jsx
<select key={version + "a"} defaultValue="cherry">
  <StaticOptions />
</select>
```

The `options` starts by being empty and the defaultValue `cherry` can't be set.
When the options is async loaded, then the first from the item is displayed.

--- 

```jsx
<select key={version + "d"} value="cherry" onChange={handleChange}>
  <AsyncOptions />
</select>
```

The `options` starts by being empty and the value `cherry` can't be set.
When the options is async loaded, then the first from the item is displayed.

The information that the options has been updated and loaded is not notified to the "parent" component. The value rendering will first update on next render.

I have made a minor adjustment your code example and applied "notify parent when the options is loaded"
https://codesandbox.io/s/select-element-with-static-value-but-async-options-0nx1p

Here the `value` gets updated as the "parent" is notified that the options has been updated.
And `defaultValue` doesn't get updated as expected because `defaultValue` is for uncontrolled components where React won't update on next renders.
